drivers/sensor/si1133/si1133.c

```diff
@@ -0,0 +1,392 @@
+	set_op[1] = SI1133_CMD_REG_PRM_SET_PRFX | chan_config->adcconfig_addr;
+	rv = param_tbl_access_op(drv_data, SI1133_I2C_REG_HOSTIN0, set_op,
+				 sizeof(set_op), &verify_op);
+	if (rv) {
+		return rv;
+	}
+	if (verify_op != set_op[0]) {
+		return -EIO;
+	}
+
+	/* set up ADCSENS for this channel */
+	set_op[0] = chan_config->adcsens_val;
+	set_op[1] = SI1133_CMD_REG_PRM_SET_PRFX | chan_config->adcsens_addr;
+	rv = param_tbl_access_op(drv_data, SI1133_I2C_REG_HOSTIN0, set_op,
+				 sizeof(set_op), &verify_op);
+	if (rv) {
+		return rv;
+	}
+	if (verify_op != set_op[0]) {
+		return -EIO;
+	}
+
+	/* set up ADCPOST for this channel */
+	set_op[0] = chan_config->adcpost_val;
+	set_op[1] = SI1133_CMD_REG_PRM_SET_PRFX | chan_config->adcpost_addr;
+	rv = param_tbl_access_op(drv_data, SI1133_I2C_REG_HOSTIN0, set_op,
+				 sizeof(set_op), &verify_op);
+	if (rv) {
+		return rv;
+	}
+	if (verify_op != set_op[0]) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int initiate_measurements(struct si1133_data *const drv_data)
+{
+	uint8_t cmd_ctr, irq_status, data[SI1133_CFG_TOTAL_OUTPUT_BYTES];
+	uint16_t uv_light_raw;
+	int rv;
+
+	rv = reset_and_read_cmd_ctr(drv_data, &cmd_ctr);
+	if (rv) {
+		return rv;
+	}
+
+	rv = si1133_write_i2c_reg(drv_data, SI1133_I2C_REG_COMMAND,
+				  SI1133_CMD_REG_FORCE);
+	if (rv) {
+		LOG_ERR("initiate measurements failed: 0x%x\n", rv);
+		return rv;
+	}
+
+	rv = poll_cmd_ctr_increment(drv_data, cmd_ctr);
+	if (rv) {
+		return rv;
+	}
+
+	irq_status = 0;
+	while (!irq_status) {
+		rv = si1133_read_i2c_reg(drv_data, SI1133_I2C_REG_IRQ_STATUS,
+					 &irq_status);
+		if (rv) {
+			LOG_ERR("read IRQ_STATUS register failed: 0x%x\n", rv);
+			return rv;
+		}
+	}
+
+	rv = i2c_burst_read(drv_data->i2c_master, drv_data->i2c_addr,
+			    SI1133_I2C_REG_HOSTOUT_BASE, data,
+			    SI1133_CFG_TOTAL_OUTPUT_BYTES);
+	if (rv) {
+		LOG_ERR("read HOSTOUTx registers failed: 0x%x\n", rv);
+		return rv;
+	}
+#ifdef CONFIG_SI1133_OPERATING_MODE_BL
+	drv_data->data.visible_light =
+	    (data[0] << 16) | (data[1] << 8) | data[2];
+	drv_data->data.infrared_light =
+	    (data[3] << 16) | (data[4] << 8) | data[5];
+	uv_light_raw = (data[6] << 8) | data[7];
+#else
+	drv_data->data.visible_light = (data[0] << 8) | data[1];
+	drv_data->data.infrared_light = (data[2] << 8) | data[3];
+	uv_light_raw = (data[4] << 8) | data[5];
+#endif
+	/* convert the sensor's raw UV measurement into UVI scale */
+	drv_data->data.ultraviolet_light =
+	    0.0187 * ((0.00391 * SQUARE(uv_light_raw)) + uv_light_raw);
+
+	return 0;
+}
+
+static int si1133_sample_fetch(const struct device *dev,
+			       enum sensor_channel chan)
+{
+	struct si1133_data *drv_data = dev->data;
+
+	return initiate_measurements(drv_data);
+}
+
+static int si1133_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	struct si1133_data *drv_data = dev->data;
+
+	switch (chan) {
+	case SENSOR_CHAN_LIGHT:
```

Comment:
`SENSOR_CHAN_LIGHT` and `SENSOR_CHAN_IR` are documented to return values as Lux, whereas it appears you are returning the raw photodiode ADC readings. These are not the same thing.